Dvclive api reference (#2632)
iterative/dvc.org: content/docs/dvclive/api-reference/log.md

# dvclive.log()

Generates _plot metrics_ by saving the given `name`: `val` pair to a `.tsv`
file.

```py

>  def log(name: str, val: Union[int, float], step: int = None) -> None:

Reviewer: Another really minor thought: In the dvc api docs, we omit the return type. Not sure if there's a style preference here, just a thought on consistency. cc @skshetry?

Author: I saw the return type being used in get_url and followed that:

https://dvc.org/doc/api-reference/get_url

I should have probably checked the other pages for consistency

# dvclive.log()

Generates _metrics logs_ (usable by `dvc plots`) by saving the given `name`:
`val` pair to a `.tsv` file.

```py
 def log(name: str, val: float, step: int = None):
```

#### Usage:

```py
import dvclive

dvclive.log("loss", 0.9)
```

## Description

The first call to `dvclive.log(name, val)` will create a new file in

> `{path}/{name}.tsv` including the header and first row.

Reviewer 2: What is `path` though? Hasn't been explained in this doc. Maybe just link to `init` earlier and avoid this if it's not central to `dvclive.log`

> `{path}/{name}.tsv` including the header and first row.
> 
> For example `dvclive.log("loss", 0.9)` will create `{path}/loss.tsv`:

Reviewer 2: Also, 💅
```suggestion
`{path}/{name}.tsv` including the header and first row. For example,
`dvclive.log("loss", 0.9)` will create a `dvclive/loss.tsv` file:
```

> ```

Reviewer 2: ````suggestion
```tsv
````

timestamp step  loss

> 1623671484747 0 0.9

Reviewer 2: Why not preserve the tab characters? It's a TSV file after all.

```

Each subsequent call to `dvclive.log(name, val)` will add a new row to

> `{path}/{name}.tsv`.
> 
> The created file `{path}/{name}.tsv` is usable by `dvc plots`.

Reviewer 2: ```suggestion
the TSV file.

> 💡 Note that TSV files can be visualized with `dvc plots`.
```

> 💡 If `name` contains slashes (i.e. `train/loss`), the required subfolders will
> be created and the file will be saved inside the last subfolder (i.e.
> `{path}/train/loss.tsv`).

Reviewer 2: ```suggestion
If `name` contains slashes (e.g. `train/loss`), the required directories will
be created (e.g. `dvclive/train/loss.tsv`).
```

💡 If you call `dvclive.log` without calling
[`dvclive.init`](/doc/dvclive/api-reference/init) first, `dvclive` will
automatically initialize itself using either default values or environment

> variables (when used alongside `DVC`).

Reviewer 2: Again, should probably mention `init` (and this info) at the beginning of the **Description**. And no need to explain DVC environment here as that should be part of the `init()` ref.

Reviewer 2: Also, we currently use 💡 in a slightly [different manner](https://dvc.org/doc/user-guide/contributing/docs#general-language-guidelines). Up to you but if you want to be consistent with DVC docs for now I'd avoid the emoji here. Could be a note though (md block quote).

## Parameters

> - `name` - The _metrics logs_ will be saved in `{path}/{name}.tsv`.

Reviewer 2: - Start with lower case (`- the ...`).
- Not sure about "metrics logs" term
- `path` is out of context.
- No period `.` needed (not a sentence)

> - `val` - The value to be added in the `name` column of a new row.

Reviewer 2: ```suggestion
- `val` - the value to add under the `name` column of a new TSV row
```

- `step` (`None` by default) - The value to be added in the `step` column of a

>   new row. If `None`, the value of the internal `_step` property will be used.

Reviewer 2: I feel like `step` could be explained or at least mentioned in the **Description**? Comes a bit out of the blue here.

Reviewer 2: Also, I'd avoid term "value" here (can be confused with `val`)
